src/features/profile/hooks/usePersonSearch.ts

```diff
@@ -0,0 +1,66 @@
+import { useApiClient } from 'core/hooks';
+import useDebounce from 'utils/hooks/useDebounce';
+import { ZetkinPerson } from 'utils/types/zetkin';
+import { useEffect, useState } from 'react';
+
+type UsePersonSearchReturn = {
+  isLoading: boolean;
+  results: ZetkinPerson[];
+  setQuery: (q: string) => void;
+};
+
+// TODO: Use search store instead, which will require refactoring by match type
+export default function usePersonSearch(orgId: number): UsePersonSearchReturn {
+  const apiClient = useApiClient();
+  const [isTyping, setIsTyping] = useState(false);
+  const [queryString, setQueryString] = useState('');
+  const [loadingByQueryString, setLoadingByQueryString] = useState<
+    Record<string, boolean>
+  >({});
+  const [matchesByQueryString, setMatchesByQueryString] = useState<
+    Record<string, ZetkinPerson[]>
+  >({});
+
+  const debouncedFinishedTyping = useDebounce(async () => {
```

Comment:
I would like to understand better in which context has sense to use the debounce method, it feels kind of tricky :/

Reply:
Great question! Debouncing is the process of avoiding to do something _too often_. In this case, every time the user types a letter, the search string changes.

But there's no use actually searching (i.e. making requests to the API) for every letter, because chances are, within milliseconds of the user typing "Ri" they're going to type another letter making "Ric", and within milliseconds of that, another letter making "Rich".

It's better to hold off and to search (i.e. make the request) only once the user stops typing, in this case defined as no changes for 600ms.

This means, that the while the user is typing, no requests are made. Once they pause for 600ms (or more) the search happens.